src/Tickets/Commerce/Order_Modifiers/Checkout/Abstract_Fees.php

```diff
@@ -0,0 +1,339 @@
+<?php
+/**
+ * Fees Class for handling fee logic in the checkout process.
+ *
+ * This class is responsible for managing fees, calculating them, and appending them
+ * to the cart. It integrates with various filters and hooks during the checkout process.
+ *
+ * @since TBD
+ * @package TEC\Tickets\Commerce\Order_Modifiers\Checkout
+ */
+
+namespace TEC\Tickets\Commerce\Order_Modifiers\Checkout;
+
+use TEC\Tickets\Commerce\Utils\Value;
+use TEC\Tickets\Commerce\Order_Modifiers\Controller;
+use TEC\Tickets\Commerce\Order_Modifiers\Modifiers\Modifier_Manager;
+use TEC\Tickets\Commerce\Order_Modifiers\Modifiers\Modifier_Strategy_Interface;
+use TEC\Tickets\Commerce\Order_Modifiers\Repositories\Fees as Fee_Repository;
+use TEC\Tickets\Commerce\Order_Modifiers\Repositories\Order_Modifier_Relationship;
+use TEC\Tickets\Commerce\Order_Modifiers\Values\Currency_Value;
+use TEC\Tickets\Commerce\Order_Modifiers\Values\Precision_Value;
+use Tribe__Template as Template;
+
+/**
+ * Class Fees
+ *
+ * Handles fees logic in the checkout process.
+ *
+ * @since TBD
+ */
+abstract class Abstract_Fees {
+
+	/**
+	 * The default priority and number of arguments for hooks.
+	 *
+	 * @var array
+	 */
+	protected array $hook_args = [
+		'ten_three' => [ 10, 3 ],
+		'ten_two'   => [ 10, 2 ],
+	];
+
+	/**
+	 * The modifier type used for fees.
+	 *
+	 * @since TBD
+	 * @var string
+	 */
+	protected string $modifier_type = 'fee';
```

Comment:
Why are we becoming so specific in an abstract ? I would prefer here only defining the property and setting it in a class extending the abstract.